derive/src/pyclass.rs

```diff
@@ -359,19 +359,19 @@ where
     Item: ItemLike + ToTokens + GetIdent,
 {
     fn gen_impl_item(&self, args: ImplItemArgs<'_, Item>) -> Result<()> {
-        let ident = if args.item.is_function_or_method() {
-            Ok(args.item.get_ident().unwrap())
-        } else {
-            Err(self.new_syn_error(args.item.span(), "can only be on a method"))
-        }?;
+        let func = args
+            .item
+            .function_or_method()
+            .map_err(|_| self.new_syn_error(args.item.span(), "can only be on a method"))?;
+        let ident = &func.sig().ident;
 
         let item_attr = args.attrs.remove(self.index());
         let item_meta = MethodItemMeta::from_attr(ident.clone(), &item_attr)?;
         let py_name = item_meta.method_name()?;
 
-        let sig_doc = args.item.function_or_method_impl().ok().map(|item| {
-            let sig = item.sig();
-            let args: Vec<_> = sig
+        let sig_doc = {
```

Comment:
this seems likes its based on an old revision. With these new changes though, `get_sig` might be better suited taking a `Signature` immediately (and `gen_module_items` for `FunctionItem` can easily be changed to do pass the sig).

Reply:
thanks! I changed it to take &Signature during rebase